File: problemsets/Codeforces/Python/A344.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
r,a=0,'a'
for _ in range(int(input())):
  b=input(); r+=a!=b; a=b
print(r)

File: problemsets/Codeforces/Javascript/A344.js
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
n = parseInt(readline()), r = 0
v = ""
while (n--) {
	u = readline()
	if (u != v)
		v = u, r++
}
print(r)
